Update go-client to user new API client with tsh profile loader. (#6310)
gravitational/teleport: examples/go-client/README.md

### Introduction

This program demonstrates how to...

> 1. Authenticate the client using user-friendly credential loaders.

Reviewer: ```suggestion
1. Authenticate the client using credential loaders.
```

2. Authorize API calls using an independent user and role.
3. Create a new client and make API calls.

### Authentication

Auth API clients must perform two-way authentication using x509 certificates to:

1. Validate the auth server x509 certificate to make sure the API endpoint can be trusted.
2. Offer their x509 certificate, which has been previously issued by the auth sever.

> Authentication is simple with the Credential loaders available in the `client` package. These loaders can be used to find and load certificates generated by `tctl` and `tsh` commands.

Reviewer: ```suggestion
Credential loaders from the `github.com/gravitataional/teleport/api/client` package can be used to find and load certificates generated by `tctl` and `tsh` commands.
```

### Authorization

The server will authorize requests for the user associated with the certificates used to authenticate the client. Therefore, to use the API client, you need to create a user and any roles it may need for your use case. 

The client will act on behalf of that user and have access as defined by the user's role(s). It is recommended to create an independent user and role to manage API access control.

### Demo

This demo can be used to quickly get the API client up and running.

##### Create resources

Create the `access-admin` user and role using the following commands:

```bash
$ tctl create -f access-admin.yaml
$ tctl users add access-admin --roles=access-admin
```

##### Generate Credentials

> This demo uses the Profile credential loader. Login with `tsh` and the Client will automatically use the generated credentials and retrieve the cluster's reverse tunnel address from the web proxy.

Reviewer: ```suggestion
This demo uses the Profile credential loader. Login with `tsh` and the Client will use the credentials from the profile directory (`~/.tsh`).
```

```bash
# login and automatically generate keys
$ tsh login --user=access-admin
```

> NOTE: You can pass the `InsecureAddressDiscovery` in `client.Config` field to skip TLS during address retrieval. This is not recommended for production clients.

Reviewer: ```suggestion
NOTE: You can pass the `InsecureAddressDiscovery` in `client.Config` field to skip verification of the TLS certificate of the proxy. This is not recommended for production clients.
```

##### Run

```bash
$ go run main.go
```

> To see more information on the Go Client and how to use it, visit our [API Documentation](https://goteleport.com/teleport/docs/api-reference/).

Reviewer: ```suggestion
To see more information on the Go Client and how to use it, visit our [API Documentation](https://pkg.go.dev/github.com/gravitational/teleport/api/client).
```

Author: Sasha is going to reroute the api-reference page to the Go-Doc, and in the future we will likely add it back in a stripped down version, which will link to the Go-Doc at the end. I'd rather keep this link as is rather than alternate it between the two as changes are made.
